src/main/healthCheck.ts

```diff
@@ -48,3 +48,23 @@ const isUrlReachable = async (url: string) => {
       })
   })
 }
+
+export const startHealthCheckPolling = () => {
+  setInterval(async () => {
+    // Don't check if proxy is offline or starting
+    // Don't check if a recording is in progress
+    // Don't check if not in the /recorder page
+    if (
+      !k6StudioState.currentProxyProcess ||
+      ['starting', 'offline'].includes(k6StudioState.proxyStatus) ||
+      k6StudioState.currentBrowserProcess ||
+      !k6StudioState.currentClientRoute.startsWith('/recorder')
```

Comment:
We could have the health check working across all routes, but I think it'd be too much, so I restricted it to these checks instead.

This means that if there's something wrong with the proxy connection, the user won't be able to know until they go to the Recorder page, which is okay for now since it's the only place using the proxy.